python/ql/src/experimental/semmle/python/frameworks/NoSQL.qll

```diff
@@ -34,40 +38,36 @@ private module NoSql {
    * Gets a reference to an initialized `Mongo` instance.
    * See `pyMongo()`, `flask_PyMongo()`
    */
-  private API::Node mongoInstance() {
+  private API::Node mongoClientInstance() {
     result = pyMongo() or
     result = flask_PyMongo()
   }
 
   /**
-   * Gets a reference to an initialized `Mongo` DB instance.
-   * See `mongoEngine()`, `flask_MongoEngine()`
-   */
-  private API::Node mongoDBInstance() {
-    result = mongoEngine().getMember(["get_db", "connect"]).getReturn() or
-    result = mongoEngine().getMember("connection").getMember(["get_db", "connect"]).getReturn() or
-    result = flask_MongoEngine().getMember("get_db").getReturn()
-  }
-
-  /**
-   * Gets a reference to a `Mongo` DB use.
-   *
-   * See `mongoInstance()`, `mongoDBInstance()`.
+   * Gets a reference to a `Mongo` DB instance.
    */
-  private DataFlow::LocalSourceNode mongoDB(DataFlow::TypeTracker t) {
+  private DataFlow::LocalSourceNode mongoDBInstance(DataFlow::TypeTracker t) {
     t.start() and
     (
       exists(SubscriptNode subscript |
-        subscript.getObject() = mongoInstance().getAUse().asCfgNode() and
+        subscript.getObject() = mongoClientInstance().getAUse().asCfgNode() and
         result.asCfgNode() = subscript
       )
```

Comment:
This could also be written as 
```ql
result.asCfgNode() = any(SubscriptNode subscript | subscript.getObject() = mongoClientInstance().getAUse().asCfgNode())
```
though it's debatable whether this is an improvement.

Also, we really ought to add support for subscripting in the API graph (as a call to `__getitem__`). It really bothers me that we have to resort to type tracking here. :cry:

Reply:
> Also, we really ought to add support for subscripting in the API graph (as a call to __getitem__). It really bothers me that we have to resort to type tracking here. cry

Yes, that would be good :+1: